rework considerations on the choice of hash algorithm (#377)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

## Choice of a Hash Algorithm

> To ensure privacy of claims that are not being selectively disclosed in a given presentation,

Reviewer: ```suggestion
To ensure privacy of claims that are selectively disclosable, but are not being disclosed in a given presentation,
```

> the hash function MUST ensure that it is infeasible to calculate the salt and claim name-value pair
> (or any portion thereof) that results in a particular digest. This implies the hash function MUST

Reviewer: `claim name-value pair` is somewhat confusing and is not used elsewhere in the spec for this purpose.

```suggestion
the hash function MUST ensure that it is infeasible to calculate one or more of the three elements
(salt, claim name, claim value) that results in a particular digest. This implies the hash function MUST
```

Reviewer 2: ```suggestion
the hash function MUST ensure that it is infeasible to calculate any portion of the three elements
(salt, claim name, claim value) from a particular digest. This implies the hash function MUST
```

be preimage resistant, but should also not allow an observer to infer any partial information about
the undisclosed content. In the terminology of cryptographic commitment schemes, the hash function
MUST be computationally hiding.

To ensure the integrity of selectively disclosable claims, the hash function MUST be second-preimage

> resistant. That is, for any salt and claim name-value pair, it is infeasible to find a different salt
> and claim name-value pair that result in the same digest.

Reviewer: ```suggestion
resistant. That is, for any combination of salt, claim name and claim value, it is infeasible to find a different combination of salt, claim name and claim value that result in the same digest.
```

The hash function SHOULD also be collision resistant. Although not essential to the anticipated uses of
SD-JWT, without collision resistance an Issuer may be able to find multiple disclosures that have the

> same hash value. The signature over the SD-JWT would not then commit the Issuer to the contents of the
> JWT, which is surprising. Where this is a concern, the collision resistance of the hash function SHOULD

Reviewer: `, which is surprising` felt out of place, at least to me.

```suggestion
same hash value. In which case, the signature over the SD-JWT would not then commit the Issuer to the contents of the
JWT. Where this is a concern, the collision resistance of the hash function SHOULD
```
